core/silkworm/types/account.cpp

```diff
@@ -84,85 +84,96 @@ size_t Account::encoding_length_for_storage() const {
     return len;
 }
 
-std::pair<uint64_t, rlp::DecodingResult> extract_incarnation(ByteView encoded) {
-    uint8_t field_set = encoded[0];
-    size_t pos{1};
-
-    if (field_set & 1) {
-        pos += encoded[pos++];
-    }
-    if (field_set & 2) {
-        pos += encoded[pos++];
-    }
-    if (field_set & 4) {
-        // Incarnation has been found.
-        uint8_t len = encoded[pos++];
-        const std::optional<uint64_t> incarnation{endian::from_big_compact<uint64_t>(encoded.substr(pos, len))};
-        if (incarnation == std::nullopt) {
-            return {0, rlp::DecodingResult::kOverflow};
-        }
-        return {*incarnation, rlp::DecodingResult::kOk};
-    }
-    return {0, rlp::DecodingResult::kOk};
-}
-
-std::pair<Account, rlp::DecodingResult> decode_account_from_storage(ByteView encoded) noexcept {
+std::pair<Account, rlp::DecodingResult> Account::from_encoded_storage(ByteView encoded_payload) noexcept {
     Account a{};
-    if (encoded.empty()) {
+    if (encoded_payload.empty()) {
         return {a, rlp::DecodingResult::kOk};
+    } else if (encoded_payload.length() == 1) {
+        // Must be at least 2 bytes : field_set + len of payload
```

Comment:
That's not right: `encode_for_storage` on an empty account will return a string of length 1 with a single 0 byte.

Reply:
Ok skimmed over it. The check has to be hardened so if `encoded_payload[0] != 0` then there MUST be something afterwards. Doing now

Reply:
The simplest thing is to remove this check altogether. The length is already checked in the main loop.

Reply:
Done 8425b9b